### DISCUSSION TITLE
Hugo FixIt 配置总览 | FixIt

### DISCUSSION THREAD

**Discussion by giscus** [hugo-fixit/FixIt#155] [General]
# Hugo FixIt 配置总览

了解如何配置你的 Hugo FixIt 站点。

https://fixit.lruihao.cn/zh-cn/docs/getting-started/configuration/introduction/

**Comment by luqicoder** [comment]
```
[params]
  # FixIt 主题版本
  version = "0.2.x"
```

我把`0.2.x`修改为：`0.2.14`之后，构建失败.
```
兼容性错误 (0.2.14 -> 0.2.X):
您进行了一次不兼容的更新。
```

**Reply by Lruihao** [reply]
这里不是填具体版本，只填中版本号即可，这是之前LoveIt 针对 `v0.1.X` 版本设计的参数

```
# FixIt 主题版本
  version = "0.2.X"
```

**Reply by Lruihao** [reply]
从 `v0.2.15` 版本开始，支持填写模糊或者具体的版本号，例如：`0.2.x`, `v0.2.15`, `0.2.15-RC` 等

**Comment by luqicoder** [comment]
```
强烈建议你把：
apple-touch-icon.png (180x180)
favicon-32x32.png (32x32)
favicon-16x16.png (16x16)
mstile-150x150.png (150x150)
android-chrome-192x192.png (192x192)
android-chrome-512x512.png (512x512)
```
我放在static目录下，发现网站图标并没有更新

**Reply by Lruihao** [reply]
如果文件名称都没错，应该就是缓存问题

**Comment by hzyangjc** [comment]
请问修改全局字体应该在哪个文件修改？还是有很多地方都要改？
想改一下英文字体

**Reply by Lruihao** [reply]
见 [3.3 自定义样式](https://fixit.lruihao.cn/zh-cn/theme-documentation-basics/#33-%E8%87%AA%E5%AE%9A%E4%B9%89%E6%A0%B7%E5%BC%8F)

相关讨论：

- https://github.com/hugo-fixit/FixIt/discussions/80
- https://github.com/hugo-fixit/FixIt/discussions/159

**Reply by hzyangjc** [reply]
哇回复好快 已解决，谢谢！

**Comment by zyyaria** [comment]
请问怎么引入fuse搜索啊？😅😅

**Reply by Lruihao** [reply]
我有空加一下fuse

**Reply by Lruihao** [reply]
已增加 fuse 搜索

**Comment by dingyufan** [comment]
为啥我这加了配置，giscus出不来。
js里断点，看到initComment方法内，this.config.comment.enable值任为false。
配置如下：
```toml
  [params.page.comment]
  enable = true
    # giscus
    [params.page.comment.giscus]
	enable = true
	repo = "dingyufan/dingyufan.github.io"
	repoId = "R_kgDOHJkJ0w"
	category = "General"
	categoryId = "DIC_kwDOHJkJ084CSMrg"
	mapping = "url"
	reactionsEnabled = "1"
	emitMetadata = "0"
	inputPosition = "top"
	lightTheme = "light"
	darkTheme = "dark"
	lazyLoad = false
```

**Reply by Lruihao** [reply]
本地开发环境不启用评论，使用生产环境即可

```
hugo server -e production
```

**Reply by zyyaria** [reply]
为啥你们俩的聊天我会收到 E-mail 的？😂

**Reply by Lruihao** [reply]
> 为啥你们俩的聊天我会收到 E-mail 的？😂

取消订阅这篇 discussion 应该就收不到了。

**Reply by dingyufan** [reply]
我不是在本地预览，是直接放到github pages上。
刚刚我也本地尝试了hugo server -e production，也还是没有。
我的配置如此 [config.toml](https://github.com/dingyufan/dingyufan.github.io/blob/main/config.toml)，不知是否有错误？

**Reply by Lruihao** [reply]
不知道是不是跟你的 config 的写的缩进错误有关。你先把主题的 [hugo.toml](https://github.com/hugo-fixit/FixIt/blob/master/hugo.toml) 复制到自己目录，然后再改试试

**Reply by dingyufan** [reply]
奇怪。 我使用了(当前站点config.toml)[https://github.com/hugo-fixit/docs/blob/main/config.toml]，本地hugo server -e production，也还是没有。
会和hugo版本有关吗？

**Reply by Lruihao** [reply]
和 hugo 版本应该没关系

**Reply by travisbikkle** [reply]
多语言的博客需要加上语言。因为FixIt的网站没加这个，可能大多数人就直接忽略了
lang = "zh-CN" 等等
不加的话，浏览器会报错CSP，跨站请求到giscus.app会被拒绝

**Comment by Raydon10** [comment]
项目根目录和主题目录都有config.toml，上面教程里的配置参数放在哪里不清不楚，感觉说的好乱，参数和现在版本的也不太一致。

**Reply by Lruihao** [reply]
> 优先级：项目的配置 >= 主题配置

主题根目标的 config 是主题的默认值，项目根目录的配置是 hugo + 主题配置，这些基本的概念，建议你还是看看 hugo 官网的入门吧。

---

参数不一样

1. 文档中包含了，历史版本的删除或更改一些参数 （有标志）
2. 主题配置文件中的只保留最新版本支持的参数，及其默认值

**Reply by Raydon10** [reply]
OK，我刚接触hugo，明白你的意思了。但这篇文章看似详细，其实对刚接触hugo的人来说就不友好了。

**Reply by Lruihao** [reply]
主题文档的目的不是教人入门 hugo, 只为了说明主题的功能即可。必要的前驱知识要自己先看先学。

**Comment by lovelxc** [comment]
请问如何加文章页面的目录，似乎默认设置了文章目录，但我这边仍然无法显示。
hugo 是0.106.0 extended版本的

**Reply by lovelxc** [reply]
好吧，找到原因了，我用的edge(107.0.1418.52)无法显示，但是谷歌可以，就很奇怪，edge开f12都找不到目录的html元素

**Comment by medivh63** [comment]
请求如何修改文本的宽度，相对比LoveIt的节目 FixIt要宽的多

**Reply by Lruihao** [reply]
页面宽度见 [3.3.1 页面宽度](https://fixit.lruihao.cn/zh-cn/documentation/advanced/#page-style)，其他样式可在 _custom.scss 中自定义

**Reply by medivh63** [reply]
非常感谢

**Reply by medivh63** [reply]
<a href="https://imgur.com/DVUAyQT"><img src="https://i.imgur.com/DVUAyQT.jpg" title="source: imgur.com" /></a>


telegram 的评论框现在和整个屏幕同宽，虽然我在配置项设置宽度，但是不起作用。 请教下我可以在哪里设置？抱歉我的前端技术太渣了

**Reply by medivh63** [reply]
怎么设置可以文本宽同宽

**Comment by feynbin** [comment]
在文章标题处的作者头像加载太慢了，能不能设置缓存在本地，可以给个示例吗。

**Reply by Lruihao** [reply]
见 https://github.com/hugo-fixit/FixIt/discussions/261

**Comment by kensou90** [comment]
你好，我想几问个关于gitalk集成的问题：
1、gitalk的浅色暗色主题怎么设置，还有怎么跟随博客主题自动更改 浅色/暗色
2、gitalk样式怎么设置的，是不是跟gitalk版本有关系，可以设置gitalk版本吗

**Reply by debuginn** [reply]
我看是默认自动跟随

**Reply by Lruihao** [reply]
gitalk 本身没有night主题，所以没办法适配。

**Comment by homeproject7** [comment]
``` toml
# FixIt 0.2.0 | 新增 代码配置
    [params.page.code]
      # 是否显示代码块的复制按钮
      copy = true
      # FixIt 0.2.13 | 新增 是否显示代码块的编辑按钮
      edit = true
      # 默认展开显示的代码行数
      maxShownLines = 10
```
请问代码块的复制与折叠，是否需要另外的配置，按照文档配置，没有效果

**Reply by Lruihao** [reply]
需要配置 markup 字段，见基础篇或者 [#43](https://github.com/hugo-fixit/FixIt/issues/43)

**Comment by woniu336** [comment]
发现个搜索的bug,  比如我搜索:hugo,出来好多内容,当鼠标滚轮往下滑的时候搜索框就收回去了.想找点内容又得重新搜一次,而且滚轮不敢动.

**Reply by Lruihao** [reply]
这是当时设计故意为之，看来这不是一个好设计:( 之后会重构搜索这个功能，改善诸多的体验问题。

**Reply by ddjj97** [reply]
+1，也遇到了这个搜索问题，鼠标下移，乱动很容易退出搜索框，比较影响搜索体验

**Comment by pihaoxuan** [comment]
为什么我根据这篇文章的主题配置，很多功能都不能实现？，好奇怪

**Reply by pihaoxuan** [reply]
跟您提供的完整配置的预览相差非常大

**Reply by Lruihao** [reply]
很多功能默认是关闭的，需要自己调整配置开启。有哪些差距呢，可以举些例子吗？

**Comment by YouZhiZheng** [comment]
请问如何可以去掉文章标题前面的绿色箭头？

**Reply by Lruihao** [reply]
去掉 front matter 中的 repost 相关参数。

**Reply by YouZhiZheng** [reply]
请问一下，我将 hugo.toml 以及front matter 中的 lightgallery 参数均设置为了 true，为什么还是不能放大图片？如果要实现放大图片这一功能我还应该怎么做？

**Reply by Lruihao** [reply]
lightgallery 设置为 "force" 或者文章中写图片时加上 title，例如：`![alt](xxx.jpg "title")`

**Reply by YouZhiZheng** [reply]
已解决，十分感谢。

**Comment by ghost** [comment]
请问，在assets/css/_override.scss里定义的全局字体变量中新引入的字体（很小众的）要包含在哪个目录下？

**Reply by Lruihao** [reply]
请看[进阶篇](https://fixit.lruihao.cn/zh-cn/documentation/advanced/#自定义样式)自定义样式，可以在 `assets/css/_custom.scss` 中自定义字体。字体文件可以放在 `static` 目录，例如 `static/fonts`。类似问题回答：[#80](https://github.com/orgs/hugo-fixit/discussions/80), [#159](https://github.com/orgs/hugo-fixit/discussions/159)。

**Reply by ghost** [reply]
ttf 字体30多MB太大了，Github  Push 不上去，换woff2字体18MB本地预览无效果，请有什么压缩/拆分或者提取部分字体的办法吗

**Reply by Lruihao** [reply]
> ttf 字体30多MB太大了，Github Push 不上去，换woff2字体18MB本地预览无效果，请有什么压缩/拆分或者提取部分字体的办法吗

可以使用类似 [font-spider](https://github.com/aui/font-spider) 等类似的工具压缩字体，我以前写了一篇教程可以看看，[ttf 字体压缩](https://lruihao.cn/posts/web-font/)。

**Reply by ghost** [reply]
谢谢，我改天试试

**Reply by ghost** [reply]
谢谢，我能把字体压到3M多，并用拆包技术（每个包拆到70k），在本地预览效果很好，不过push到github上就加载不了css了，根据[拆包软件给的信息](https://github.com/KonghaYao/cn-font-split/blob/main/README.md),可能是Github做了并发数限制，就用了[字图CDN](https://chinese-font.netlify.app/cdn/)，流畅多了

**Reply by ghost** [reply]
这还有css预加载的问题

**Comment by ghost** [comment]
有没有什么办法，使主页显示的文章不止包含posts目录下的，也包含其他目录下的？

**Reply by Lruihao** [reply]
其他目录的文章 front matter 指定 type: posts

**Reply by ghost** [reply]
还有就是，全局字体设置对评论系统无效吗？

**Reply by Lruihao** [reply]
> 还有就是，全局字体设置对评论系统无效吗？

无效，评论系统是一个单独的 iframe 由每个单独评论系统控制。

**Comment by Rookiecom** [comment]
请问开启搜索功能只需要这样吗，我这样设置之后搜索时会卡住，搜索不出结果
[params.search]
  enable = true

**Reply by Lruihao** [reply]
文档中有写（[search](https://fixit.lruihao.cn/zh-cn/documentation/getting-started/configuration/#search)），还需要添加 outputs 配置：

> 注：0.3.9 版本之前搜索的输出配置为 `JSON`, 0.3.10 开始输出配置改为 `search`，根据你的 FixIt 版本配置 outputs 配置即可。

<img width="819" alt="image" src="https://github.com/user-attachments/assets/f58f031f-cd75-4e45-838e-7ecd0fffdf9d">

**Reply by Rookiecom** [reply]
我是有认真看文档，也看到outputs这一行的。
我是采用覆盖的方法配置的：
```toml
[markup]
  _merge = "shallow"
```
然后我看主题默认的配置中有这样的：

```toml
[outputs]
  home = ["html", "rss", "archives", "search"]
  page = ["html", "markdown"]
  section = ["html", "rss"]
  taxonomy = ["html"]
  term = ["html", "rss"]
```
所以我就没有添加在我自己的配置文件中。
但是，你回复之后我添加了，发现就好起来了。不管什么，感谢你的帮助。

另外，我还想问一个非常小白的问题，如何全局更换字体，我对前端一窍不通。
下面是我的情况说明：
我想通过[字图CDN](https://chinese-font.netlify.app/fonts/lxgwwenkai/LXGWWenKai-Regular)来使用霞鹭文楷字体，我看了所有有关字体的issue和文档，但是因为每个地方都不是很统一（自我感觉），我试了一通都没有成功。我的做法如下：

在_override.scss中添加：
```scss
@import url('https://chinese-font.netlify.app/fonts/lxgwwenkai/LXGWWenKai-Regular');
$global-font-family: lxgwwenkai,system-ui, -apple-system, BlinkMacSystemFont, PingFang SC, Microsoft YaHei UI, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, Helvetica, Arial, sans-serif !default;
```
但是没能生效

**Reply by Rookiecom** [reply]
另外，我还尝试查看了是否支持callout，因为我看文档经常有这样的块。我指的callout：

> [!info]
> Here's a callout block.
> It supports **Markdown**, [[Internal link|Wikilinks]], and [[Embed files|embeds]]!

**Reply by Lruihao** [reply]
> 另外，我还尝试查看了是否支持callout，因为我看文档经常有这样的块。我指的callout：
> 
> > [!info]
> > Here's a callout block.
> > It supports **Markdown**, [[Internal link|Wikilinks]], and [[Embed files|embeds]]!

这种写法暂时不支持。

**Reply by Lruihao** [reply]
> 另外，我还想问一个非常小白的问题，如何全局更换字体，我对前端一窍不通。
> 下面是我的情况说明：
> 我想通过[字图CDN](https://chinese-font.netlify.app/fonts/lxgwwenkai/LXGWWenKai-Regular)来使用霞鹭文楷字体，我看了所有有关字体的issue和文档，但是因为每个地方都不是很统一（自我感觉），我试了一通都没有成功。我的做法如下：
> 
> 在_override.scss中添加：
> ```scss
> @import url('https://chinese-font.netlify.app/fonts/lxgwwenkai/LXGWWenKai-Regular');
> $global-font-family: lxgwwenkai,system-ui, -apple-system, BlinkMacSystemFont, PingFang SC, Microsoft YaHei UI, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, Helvetica, Arial, sans-serif !default;
> ```
> 但是没能生效

我看它的字体声明是 'LXGW WenKai'，而不是 lxgwwenkai。另外就是你的 `@import` 的 css 链接也是错的，$global-font-family 再次被定义时其后不需要加 `!default`。根据 [字图CDN](https://chinese-font.netlify.app/fonts/lxgwwenkai/LXGWWenKai-Regular) 的引入指引修正这些错误后的代码如下：

```scss
@import url('https://chinese-fonts-cdn.deno.dev/packages/lxgwwenkai/dist/LXGWWenKai-Regular/result.css');
$global-font-family: 'LXGW WenKai', system-ui, -apple-system, BlinkMacSystemFont, PingFang SC, Microsoft YaHei UI, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, Helvetica, Arial, sans-serif;
```

<img width="581" alt="image" src="https://github.com/user-attachments/assets/d04885fb-7279-4f94-a0e6-7bca0bbd442a">

**Reply by Rookiecom** [reply]
感谢你的回答！！！！！！！

**Reply by Lruihao** [reply]
> 另外，我还尝试查看了是否支持callout，因为我看文档经常有这样的块。我指的callout：

目前 0.3.13 版本已经完全支持 callout 以及 GitHub alert 写法了。

**Reply by GoranTan** [reply]
写的真棒，好用，比自己下载包再定义应用方便很多

**Comment by yoyoshenTW** [comment]
目前我在 content 底下直接放 .md 檔案
我希望能在 browser 上顯示內容時，能夠佔滿整個 browser 的寬度，而不是限制在一個 content 的寬度
從 html 上看似乎是在一個 class=container 內
但不知道要在哪邊進行修改

**Reply by Lruihao** [reply]
看看这里  https://fixit.lruihao.cn/zh-cn/documentation/advanced/#page-style

**Reply by yoyoshenTW** [reply]
抱歉，想再跟您請教實際的操作，我剛剛進行了下反覆的操作都沒有效果
1. 將我 root 下的 hugo.toml 內的 pageStyle 做調整，都沒有生效
2. 在 themes 底下或 root 底下的 assets/css/_custom.scss 內增加範例，也都無生效

在思考是不是找錯地方或其他原因沒控制到

```
[data-page-style='custom'] {
    @media only screen and (min-width: 1441px) {
      %page-style {
        width: Round(90%, 2px);
      }
    }
  
    @media only screen and (min-width: 1440px) {
      %page-style {
        width: Round(90%, 2px);
      }
    }
  
    @media only screen and (min-width: 1200px) {
      %page-style {
        width: Round(90%, 2px);
      }
    }
  }
```

**Reply by Lruihao** [reply]
<img width="1439" alt="image" src="https://github.com/user-attachments/assets/3c3f262a-f946-414f-b371-adbadbe57150">

按照你上面寫的樣式，我試過是生效的。檢查一下你的配置是否設置為 `pageStyle = "custom"`,就像這樣：

```toml
[params]
  # 其他參數 ...
  [params.page]
    # 其他參數 ...
    pageStyle = "custom"
```

**Reply by yoyoshenTW** [reply]
解決了，結果是我custom 的檔案位置處理錯誤，感謝你！

**Comment by 0x5c0f** [comment]
你好，请教一下首页的这一块是如何配置的 https://imgur.com/sy7TbWP , 我观察了很多的案例都有，但我不知道该如何设置

**Reply by Lruihao** [reply]
配置文件里搜索一下  profile

**Reply by 0x5c0f** [reply]
已解决，感谢

**Comment by 0x5c0f** [comment]
你好，这个原始文档连接开启后，点击变成了下载文件，如何让他可以在浏览器中直接打开而不是下载呢

**Comment by Lruihao** [comment]
要让 Markdown 文件的链接在浏览器直接预览纯文本而不是下载，你需要在服务器上设置正确的 MIME 类型。以下是一些常见的服务器配置示例：

### Nginx

在 Nginx 配置文件中，确保添加以下行：

```nginx
location ~* \.md$ {
    default_type text/plain;
    add_header Content-Disposition "inline";
}
```

### Apache

在 Apache 配置文件或 `.htaccess` 文件中，添加以下行：

```apache
<FilesMatch "\.md$">
    Header set Content-Type text/plain
    Header set Content-Disposition "inline"
</FilesMatch>
```

### 注意事项

1. **重启服务器**：在修改配置后，确保重启服务器以使更改生效。
2. **测试**：在浏览器中直接访问 Markdown 文件的 URL，以确认它以文本形式显示，而不是下载。
3. _一般的第三方静态服务都不支持自定义服务器配置，比如：GitHub Pages 等_

**Reply by 0x5c0f** [reply]
好吧，谢谢，我记得这个似乎是可以在主题里面设置的，我无法调整的我服务器的mime类型，如果主题无法更改，请忽略此问题

**Comment by WeMingT** [comment]
为什么我在配置Waline评论系统时在production模式下不显示评论系统？
```
[params]
  [params.page]
    [params.page.comment]
      enable = true
      [params.page.comment.waline]
        enable = true
        serverURL = "https://xxxxxxxxxxxxxx"
        pageview = true # 
        emoji = ["//unpkg.com/@waline/emojis@1.1.0/weibo"]
        meta = ["nick", "mail", "link"]
        requiredMeta = []
        login = "enable"
        wordLimit = 0
        pageSize = 10
        imageUploader = false # 
        highlighter = false # 
        comment = true # 
        texRenderer = false # 
        search = false # 
        recaptchaV3Key = "" # 
        turnstileKey = "" # 
        reaction = false # 
```
其他几个评论系统是enable = false.
但是我在其他主题中使用同样的serverURL配置Waline评论系统却可以正常显示？

**Reply by WeMingT** [reply]
已解决。刚才又试了一下，发现竟然可以用了。

**Comment by 6am600** [comment]
大佬，创建了一个新的页面，怎么让这个页面中的图片也支持lightgallery呢，我在md文件中加上lightgallery: true无效
---
title: "闲言碎语"
date: 2025-01-23T19:00:00+08:00
draft: false
layout: "essay"  # 使用essay布局
lightgallery: true
---

**Reply by Lruihao** [reply]
如果你自定义了essay布局的话，`lightgallery: true` 这样看起来没问题，图片写法 `![alt](/images/xxx.jpg "title")`

**Comment by WeMingT** [comment]
# 如何在menu添加“合集”选项？
即在以下的menu中（我的网站），
![Image](https://github.com/user-attachments/assets/fdfeb372-066a-4aa2-94b8-13a23e45bed4)
添加以下的“合集”选项（https://fixit.lruihao.cn/zh-cn/）。
![Image](https://github.com/user-attachments/assets/7895dfd5-c8c2-4a8c-9126-aece1abe23ec)
---
另外，我发现点击以下图片带合集图标的“诗文”选项后（我的网站），
![Image](https://github.com/user-attachments/assets/41556523-7a05-4f31-a6db-2339f4f31bb4)
可以进入如下的“所有合集”页面，
![Image](https://github.com/user-attachments/assets/37a609db-877f-4eab-bec0-a7b6ab873aa5)
但是怎样才能将其添加在menu中呢？
---
另外如何使“所有合集”页面的每个合集名称下的文章按category排列并显示category名称？

**Comment by Lzm1996** [comment]
你好，想问下想把目录从右边改到左边，应该怎么配置或者修改哪个文件呀

**Reply by Lruihao** [reply]
<img width="622" alt="image" src="https://github.com/user-attachments/assets/cf242caf-557e-46b9-97c3-a31033af092b" />

**Comment by kaishuizhuhuoyu** [comment]
hello

**Reply by kaishuizhuhuoyu** [reply]
你好

**Reply by kaishuizhuhuoyu** [reply]
👊

**Comment by AtticusLv** [comment]
求助，升级了最新版本，首页上不显示posts了，只显示非posts的了，是home.html不对还是hugo.toml配置不对？

https://github.com/AtticusLv/tech-blog

**Reply by Lruihao** [reply]
设置一下 [mainSections](https://gohugo.io/methods/site/mainsections/) 参数，不设置默认是内容最多的一种，对于你的博客而言就是 `tech` 了，你可以参考主题文档的[配置](https://github.com/hugo-fixit/docs/commit/d0a6de7c86a44df567a74ee335c5d35be45ab955)，例如在 `hugo.toml` 里设置：

```toml
mainSections = ["posts", "tech"]
```

**Comment by yzxISUya** [comment]
showFullContent = false
顶部也加了一句summaryLength = 50
为什么进入首页后每篇文章还是会显示出全文（不会截断）

**Reply by yzxISUya** [reply]
解决了（在md文件手写summary就可以了），不过summary必须要在index.md手写吗？不能直接设置为默认前多少行那种吗？

**Reply by Lruihao** [reply]
https://fixit.lruihao.cn/zh-cn/documentation/content-management/introduction/#%E5%86%85%E5%AE%B9%E6%91%98%E8%A6%81

**Reply by yzxISUya** [reply]
芥末快，ok蟹蟹
